Adding Gamepad support

rippy opened the pull request:
Adding Gamepad support

clairefro commented:
Thank you!!!! I just simplified the inputs UI in `gamemaanger` that should make it easier to handle all inputs at the game manager level (and not need to worry about it in the games 

If you hav ea change, can you try editing your PR to adapt to the new input API?


```js
// maybe something in here to merge optional gamepad inputs? 
const KEY_MAPPINGS = {
  left: ["ArrowLeft", "a", "A"],
  right: ["ArrowRight", "d", "D"],
  up: ["ArrowUp", "w", "W"],
  down: ["ArrowDown", "s", "S"],
};

this.input = {
      keys: new Set(), 
      isPressedLeft: () => this.isDirectionPressed("left"),
      isPressedRight: () => this.isDirectionPressed("right"),
      isPressedUp: () => this.isDirectionPressed("up"),
      isPressedDown: () => this.isDirectionPressed("down"),
      // Legacy key check for backward compatibility...
      // NOT RECOMMENDED: CLAIRE MIGHT RMEOVE
      pressed: (key) => this.input.keys.has(key),
    };

//...

window.addEventListener("keydown", (e) => this.input.keys.add(e.key));
   window.addEventListener("keyup", (e) => this.input.keys.delete(e.key));
```

rippy commented:
> Thank you!!!! I just simplified the inputs UI in `gamemaanger` that should make it easier to handle all inputs at the game manager level (and not need to worry about it in the games
> 
> If you hav ea change, can you try editing your PR to adapt to the new input API?
> 
> ```js
> // maybe something in here to merge optional gamepad inputs? 
> const KEY_MAPPINGS = {
>   left: ["ArrowLeft", "a", "A"],
>   right: ["ArrowRight", "d", "D"],
>   up: ["ArrowUp", "w", "W"],
>   down: ["ArrowDown", "s", "S"],
> };
> 
> this.input = {
>       keys: new Set(), 
>       isPressedLeft: () => this.isDirectionPressed("left"),
>       isPressedRight: () => this.isDirectionPressed("right"),
>       isPressedUp: () => this.isDirectionPressed("up"),
>       isPressedDown: () => this.isDirectionPressed("down"),
>       // Legacy key check for backward compatibility...
>       // NOT RECOMMENDED: CLAIRE MIGHT RMEOVE
>       pressed: (key) => this.input.keys.has(key),
>     };
> 
> //...
> 
> window.addEventListener("keydown", (e) => this.input.keys.add(e.key));
>    window.addEventListener("keyup", (e) => this.input.keys.delete(e.key));
> ```

Added!  Eventually, we should make these mappings in a lookup.  These are just for my CandyCon controller, other controllers may have different button ID mappings for the D-Pad.

clairefro reviewed (approved):
Thanks!
